src/NumericalAlgorithms/RootFinding/TOMS748.hpp

```diff
@@ -54,9 +54,8 @@ double toms748(const Function& f, const double lower_bound,
             absolute_tolerance +
                 relative_tolerance * fmin(fabs(lhs), fabs(rhs)));
   };
-  // clang-tidy: internal boost warning, can't fix it.
-  auto result = boost::math::tools::toms748_solve(  // NOLINT
-      f, lower_bound, upper_bound, tol, max_iters);
+  auto result = boost::math::tools::toms748_solve(f, lower_bound, upper_bound,
```

Comment:
This change seems unrelated to the rest of the commit. Better factor it into a separate commit if that's not too painful to do.